en/components/grids_templates/row-drag.md

```diff
@@ -618,6 +618,257 @@ Notice that we also have row selection enabled and we preserve the selection whe
+
+```typescript
+class MyGridScrollComponent {
+    ...
+    private isGridScrolledToEdge(dir: 1 | -1): boolean {
+        if (this.grid.data[0] === this.grid.rowList.first.data && dir === -1) {
+            return true;
+        }
+        if (
+            this.grid.data[this.grid.data.length - 1] === this.grid.rowList.last.data &&
+            dir === 1
+        ) {
+            return true;
+        }
+        return false;
+    }
+
+    private scrollGrid(dir: 1 | -1): void {
+        if (!this.isGridScrolledToEdge(dir)) {
+            if (dir === 1) {
+                this.grid.verticalScrollContainer.scrollNext();
+            } else {
+                this.grid.verticalScrollContainer.scrollPrev();
+            }
+        }
+    }
+}
+
+```
+
+We'll still be subscribing to the `dragMove` event of the specific row in the way we did in the previous example. Since `dragMove` is fired only when the cursor actually moves, we want to have a nice and simple way to auto-scroll the grid when the row is at one of the edges, but the user **does not** move the mouse. We'll an additional method which will setup an `interval`, auto-scrolling the grid every `500ms`.
+
+We create and subscribe to the `interval` when the pointer reaches the grid's edge and we `unsubscribe` from that `interval` everytime the mouse moves or the row is dropped (regardless of cursor position).
+
+@@if (igxName === 'IgxGrid') {
+```typescript
+class MyGridScrollComponent {
+    public ngAfterViewInit() {
+        this.grid.onRowDragStart
+            .pipe(takeUntil(this.destroy$))
+            .subscribe(this.onDragStart.bind(this));
+        this.grid.onRowDragEnd
+            .pipe(takeUntil(this.destroy$))
+            .subscribe(() => this.unsubInterval());
+    }
+    ...
+    private onDragMove(event: IDragMoveEventArgs): void {
+        this.unsubInterval();
+        const dir = this.isPointOnGridEdge(event.pageY);
+        if (!dir) {
+            return;
+        }
+        this.scrollGrid(dir);
+        if (!this.intervalSub) {
+            this.interval$ = interval(500);
+            this.intervalSub = this.interval$.subscribe(() => this.scrollGrid(dir));
+        }
+    }
+    ...
+    private unsubInterval(): void {
+        if (this.intervalSub) {
+        this.intervalSub.unsubscribe();
+        this.intervalSub = null;
+        }
+    }
+}
+```
+}
+
+Following is the example of both scenarios described above - showing a drop indicator and scrolling the viewport when border's edge is reached.
+
+@@if (igxName === 'IgxGrid') {
```

Comment:
I think the whole section (so, just before "Improving UX in row drag scenarios") should be wrapped with `@@if (igxName === 'IgxGrid')`. With the current setup, you'd just have a lot of text and snippets (which we're not sure are correct) for the Tree and Hierarchical grids.
I believe that's the best solution for now - we can log an enhancement and add separate samples for Tree and Hierarchical grids later on.